### PR TITLE
PE-423 Include learning tracker report

### PR DESCRIPTION
**Pull request by andrey-canon**
## Description
- Re factor in order to use a unique general view
- Create a new task for learning tracker metrics
- Add new backends

### CODE REVIEW

File: openedx_proversity_reports/tasks.py [hunk 1]
```diff
@@ -82,3 +84,25 @@ def generate_time_spent_report(courses):
...
+        Dict with the data for every course.
+    """
+    data = {}
+
+    for course in courses:
+
+        try:
+            data[course] = LearningTrackerReport(course).generate_report()
+        except InvalidKeyError:
+            continue
```
Comment (Squirrel18) [inline openedx_proversity_reports/tasks.py]: Should we catch another Exception or maybe a broad exception? @andrey-canon 
Just not to return a 500 error.

File: openedx_proversity_reports/api/v0/views.py [hunk 2]
```diff
@@ -17,12 +17,19 @@
     generate_completion_report,
     generate_last_page_accessed_report,
     generate_time_spent_report,
+    generate_learning_tracker_report,
 )
 
 logger = logging.getLogger(__name__)
+TASKS = {
```
Comment (Squirrel18) [inline openedx_proversity_reports/api/v0/views.py]: I think this should come from the plugin settings. @andrey-canon

File: openedx_proversity_reports/api/v0/urls.py [hunk 3]
```diff
@@ -5,35 +5,20 @@
 
 from . import views
 
+REPORT_NAME_PATTERN = r'(?P<report_name>(generate)+[a-z-]+)'
+
 urlpatterns = [
+
```
[the remark this answers is not in the document]
Comment (andrey-canon) [reply openedx_proversity_reports/api/v0/urls.py]: changed

File: openedx_proversity_reports/api/v0/views.py (hunk 2 above)
Comment (andrey-canon) [reply openedx_proversity_reports/api/v0/views.py]: done

File: openedx_proversity_reports/settings/common.py [hunk 4]
```diff
@@ -50,3 +52,10 @@ def plugin_settings(settings):
     settings.OPR_SUPPORTED_FIELDS = 'openedx_proversity_reports.edxapp_wrapper.backends.supported_fields_i_v1'
     settings.OPR_GOOGLE_ANALYTICS_CREDENTIALS = {}
     settings.OPR_GOOGLE_ANALYTICS_VIEW_ID = ''
+
```
Comment (Squirrel18) [inline openedx_proversity_reports/settings/common.py]: Unnecessary break line here. @andrey-canon

File: openedx_proversity_reports/api/v0/views.py [hunk 5]
```diff
@@ -13,18 +15,15 @@
...
 from openedx_proversity_reports.edxapp_wrapper.get_openedx_permissions import get_staff_or_owner
-from openedx_proversity_reports.tasks import (
-    generate_completion_report,
-    generate_last_page_accessed_report,
-    generate_time_spent_report,
-)
 
 logger = logging.getLogger(__name__)
+BLOCK_DEFAULT_REPORT_FILTER = ['vertical']
+SUPPORTED_TASKS_MODULE = 'openedx_proversity_reports.tasks'
```
Comment (Squirrel18) [inline openedx_proversity_reports/api/v0/views.py]: If the module task will be configurable, let's put it on the plugin settings. @andrey-canon

File: openedx_proversity_reports/api/v0/views.py [hunk 6]
```diff
@@ -73,159 +71,13 @@ def post(self, request):
...
-                        }
-                     }
-                  ]
-               }
-            }
-        """
-        task_id = request.GET.get("task_id")
+        report_name = report_name.replace('-', '_')
+        task_module = import_module(SUPPORTED_TASKS_MODULE)
+        task = getattr(task_module, report_name, None)
```
Comment (Squirrel18) [inline openedx_proversity_reports/api/v0/views.py]: Getting the task module could be a unique function, maybe in the future we need to add more logic to this. @andrey-canon

File: openedx_proversity_reports/api/v0/views.py (hunk 5 above)
Comment (andrey-canon) [reply openedx_proversity_reports/api/v0/views.py]: This is not configurable, it is a simple constant

Comment (Squirrel18) [reply openedx_proversity_reports/api/v0/views.py]: Could it be configurable? @andrey-canon

Comment (andrey-canon) [reply openedx_proversity_reports/api/v0/views.py]: No, that is out of the scope

Comment (Squirrel18) [reply openedx_proversity_reports/api/v0/views.py]: In the future? just asking... @andrey-canon

Comment (andrey-canon) [reply openedx_proversity_reports/api/v0/views.py]: imo that is not necessary

File: openedx_proversity_reports/api/v0/views.py (hunk 6 above)
Comment (andrey-canon) [reply openedx_proversity_reports/api/v0/views.py]: done

File: openedx_proversity_reports/settings/common.py (hunk 4 above)
Comment (andrey-canon) [reply openedx_proversity_reports/settings/common.py]: removed

**Pull request merged**